reconcile/utils/deadmanssnitch_api.py

```diff
@@ -0,0 +1,72 @@
+import logging
+from typing import (
+    Any,
+    Optional,
+)
+
+import requests
+from pydantic import BaseModel
+
+BASE_URL = "https://api.deadmanssnitch.com/v1/snitches"
+
+
+class DeadManssnitchException(Exception):
+    pass
+
+
+class Snitch(BaseModel):
+    token: str
+    href: str
+    name: str
+    tags: list[str]
+    notes: str
+    status: str
+    check_in_url: str
+    interval: str
+    alert_type: str
+    alert_email: list[str]
+    vault_data: Optional[str]
+
+    def get_cluster_name(self) -> str:
+        return self.name.split(".")[1]
+
+    def to_dict(self) -> dict[str, Any]:
+        return self.dict(by_alias=True)
+
+
+class DeadMansSnitchApi:
+    def __init__(self, token: str, url: str = BASE_URL) -> None:
+        self.token = token
+        self.url = url
+        self.session = requests.Session()
+
+    def close_session(self) -> None:
+        self.session.close()
```

Comment:
for easy context management, use `__enter__` and `__exit__` as context manager to close session, can call `self.close_session` in `__exit__`